server.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const cowsay = require('cowsay');
+const http = require('http');
+const url = require('url');
+const querystring = require('querystring');
+const parseBody = require('./lib/parse-body.js');
+const PORT = process.env.PORT || 8080;
+
+const server = http.createServer(function(req, res) {
+  req.url = url.parse(req.url);
+  req.url.query = querystring.parse(req.url.query);
+
+  if(req.method === 'GET' && req.url.pathname === '/') {
+    res.writeHead(200, {'Content-Type': 'text/plain'});
+    return res.end('hello from my server!');
+  }
+
+  if(req.method === 'GET' && req.url.pathname === '/cowsay' && req.url.query.text !== undefined) {
+    res.writeHead(200, {'Content-Type': 'text/plain'});
```

Comment:
Another, more simple approach, would be to check that req.url.query.text exists so then you don't have to write out req.url.query.text !== undefined :) Either, or will still work the same!